skops/io/_visualize.py

```diff
@@ -232,7 +232,15 @@ def walk_tree(
             "here: https://github.com/skops-dev/skops/issues"
         )
 
-    if node_name == "constructor":
+    if isinstance(node, type):
+        yield NodeInfo(
+            level=level,
+            key=node_name,
+            val=type(node).__name__,
+            is_self_safe=False,
+            is_safe=False,
```

Comment:
here we should defer to the information in the Node instead of assuming it to be unsafe all the time. the `Tree` constructor for instance, is safe.

Reply:
so probably the parent when doing the `walk_tree` should send this info to the child.